File: kata-others/kata-python/open_parens.py
import fileinput

def main():
    for line in fileinput.input():
        stack = []
        for i in list(line):
            if not stack or stack[-1] == i: stack.append(i)
            else: stack.pop()
        if stack: print('yes')
        else: print('no')

if __name__ == "__main__":
    main()

File: kata-others/kata-rust/src/bin/open_parens.rs
use std::io::{self, BufRead};

fn main() {
    let stdin = io::stdin();
    for line in stdin.lock().lines() {
        let input = line.unwrap();
        let mut stack: Vec<char> = Vec::new();
        for c in input.chars() {
            if c == '(' {
                stack.push(c);
            } else {
                if stack.is_empty() {
                    stack.push(c);
                    break;
                } else {
                    stack.pop();
                }
            }
        }
        if stack.is_empty() {
            println!("yes");
        } else {
            println!("no");
        }
    }
}
